src/NuGet.Core/NuGet.PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -188,6 +223,48 @@ public async Task<IEnumerable<PackageRestoreData>> GetPackagesInSolutionAsync(st
             }
         }
 
+        public virtual async Task<PackageRestoreResult> RestoreMissingAssetsFileInSolutionAsync(
+            string solutionDirectory,
+            INuGetProjectContext nuGetProjectContext,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            var projects = (await SolutionManager.GetNuGetProjectsAsync()).Where(p => p is INuGetIntegratedProject);
+            var dependencyGraphContext = new DependencyGraphCacheContext(logger, Settings);
+            var originalPackageSpec = await DependencyGraphRestoreUtility.GetProjectSpec(projects.FirstOrDefault() as BuildIntegratedNuGetProject, dependencyGraphContext);
+            var providerCache = new Commands.RestoreCommandProvidersCache();
+            var now = DateTimeOffset.UtcNow;
+            void cacheModifier(SourceCacheContext cache) => cache.MaxAge = now;
+            var enabledSources = SourceRepositoryProvider.GetRepositories();
+            var allSources = new HashSet<SourceRepository>(enabledSources, new SourceRepositoryComparer());
+
+            var restoreResults = await DependencyGraphRestoreUtility.PreviewRestoreAsync(
+                SolutionManager,
+                projects.FirstOrDefault() as BuildIntegratedNuGetProject,
```

Comment:
This will be null if there is no BuildIntegratedNuGetProject. Is that handled and expected?